### PR TITLE
[Embind] Fix a closure compiler quote error

### LINKED ISSUE #15692: Embind's call() is broken by closure=1 DYNAMIC_EXECUTION=0
Repro:
```
#include <emscripten/bind.h>
#include <emscripten/val.h>
std::string Foo() {
  return emscripten::val(123).call<std::string>("toString");
}
EMSCRIPTEN_BINDINGS(bar) {
  emscripten::function("Foo", &Foo);
}
```
Compilation:
```
emcc -O2 --bind --closure=1 -s DYNAMIC_EXECUTION=0 main.cc
```
When loading this and calling `Module.Foo()`, this exception is raised:
```
a.out.js:37 Uncaught TypeError: d.C is not a function
    at a.out.js:37
    at _emval_call_method (a.out.js:36)
    at a.out.wasm:0x52c
    at a.out.wasm:0x4cc
    at a.out.wasm:0x679
    at Object.Foo2 (a.out.js:21)
    at <anonymous>:1:20
```

### PR DESCRIPTION
**Pull request by kripken**
Fixes #15692

### CODE REVIEW

File: tests/test_other.py [hunk 1]
```diff
@@ -2346,18 +2346,24 @@ def test_embind_closure_no_dynamic_execution(self):
...
         out(Module['foo'](10));
+        out(Module['bar']());
       };
     ''')
     create_file('main.cpp', r'''
+      #include <string>
       #include <emscripten/bind.h>
       int foo(int x) { return x; }
+      std::string bar() {
+        return emscripten::val(123).call<std::string>("toString");
```
Comment (sbc100) [inline tests/test_other.py:2357]: Is it not enough to just `return "123"` here?   i.e. is the fix for the JS->wasm directly or the wasm->JS direction or both?

**Review by sbc100 — APPROVED** [review]

File: tests/test_other.py (hunk 1 above)
Comment (kripken) [reply tests/test_other.py:2357]: The bug does not happen without it. I'm not familiar enough with embind details to understand why though.

Comment (sbc100) [reply tests/test_other.py:2357]: Fair enough.. I does the bug happen if you just call `emscripten::val(123).call<std::string>("toString");` within the existing bound `foo`?   i.e. do you need to declare and bind a string-returning C function to tickly this bug it is it enough to just call `"toString"` on something?

lgtm either way.

File: tests/test_other.py [hunk 2]
```diff
@@ -2346,18 +2346,26 @@ def test_embind_closure_no_dynamic_execution(self):
...
+      #include <emscripten/console.h>
       #include <emscripten/bind.h>
       int foo(int x) { return x; }
+      void bar() {
+        emscripten::val(123).call<std::string>("toString");
+        emscripten_console_log("ok");
+      }
       EMSCRIPTEN_BINDINGS(baz) {
         emscripten::function("foo", &foo);
+        emscripten::function("bar", &bar);
```
Comment (sbc100) [inline tests/test_other.py:2363]: Sees like you don't even need to declare bar or bind it .. you could just do:

```
int main() {
  emscripten::val(123).call<std::string>("toString");
  emscripten_console_log("ok");
}
```

No a big deal though..

**Pull request merged**